### DISCUSSION TITLE
How to remove ResXmlAttribute in AndroidManifest.xml

### DISCUSSION THREAD

**Discussion by HarukaMomono** [REAndroid/ARSCLib#45] [Q&A] [answered]
Hello, how can I remove an attribute from an element in the Manifest file,

I tried some ways like  onRemoved()  or  set something to null,

but I failed.

**Comment by HarukaMomono** [comment]
I found a way to remove attributes, but it may not be correct:

public static void removeAttribute(ResXmlAttribute delete) throws Exception {
    Method method = ResXmlAttribute.class.getDeclaredMethod("setNameReference", int.class);
    method.setAccessible(true);
    method.invoke(delete, -1);
    delete.getParentResXmlElement().removeUndefinedAttributes();
}

**Reply by REAndroid** [reply, marked as answer]
Use `delete.removeSelf();`

See:
[AndroidManifestHelper.java](https://github.com/REAndroid/APKEditor/blob/master/src/main/java/com/reandroid/apkeditor/common/AndroidManifestHelper.java)
